go/pkg/cas/client.go

```diff
@@ -142,11 +158,19 @@ func NewClientWithConfig(ctx context.Context, conn *grpc.ClientConn, instanceNam
 		byteStream:   bspb.NewByteStreamClient(conn),
 		cas:          repb.NewContentAddressableStorageClient(conn),
 	}
+	client.init()
 
-	// TODO(nodir): Check capabilities.
+	// TODO(nodir): Check capabilities, optionally.
 	return client, nil
 }
 
+// init is a part of NewClientWithConfig that can be done in tests without
+// creating a real gRPC connection. This function exists purely to aid testing,
+// and is tightly coupled with NewClientWithConfig.
+func (c *Client) init() {
```

Comment:
This could be in the tests package?

Reply:
it is used in `NewClient`. Could you elaborate?

Reply:
ah, at first glance I thought this would be specific for tests. Nvm.